src/test-apps/happy/tests/standalone/inet/test_inet_dns_01.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+
+
+#
+#    Copyright (c) 2016-2017 Nest Labs, Inc.
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+#
+#    @file
+#       Calls Weave Inet test between nodes.
+#
+import itertools
+import os
+import unittest
+import set_test_path
+
+from happy.Utils import *
+import happy.HappyNodeList
+import WeaveInetDNS
+import WeaveUtilities
+import subprocess
+
+count = 2
+length = 500
+
+class test_inet_dns(unittest.TestCase):
+    def setUp(self):
+        if "WEAVE_SYSTEM_CONFIG_USE_LWIP" in os.environ.keys() and os.environ["WEAVE_SYSTEM_CONFIG_USE_LWIP"] == "1":
+            self.using_lwip = True
+        else:
+            self.using_lwip = False
+
+        if self.using_lwip:
+            topology_shell_script = os.path.dirname(os.path.realpath(__file__)) + \
+                "/topology/two_nodes_on_tap_wifi.sh"
+            output = subprocess.call([topology_shell_script])
+        else:
+            topology_shell_script = os.path.dirname(os.path.realpath(__file__)) + \
+                "/topology/two_nodes_on_wifi.sh"
+            output = subprocess.call([topology_shell_script])
+
+    def tearDown(self):
+        # cleaning up
+        subprocess.call(["happy-state-delete"])
+
+    def test_inet_dns(self):
+        if "WEAVE_SYSTEM_CONFIG_USE_LWIP" in os.environ.keys() and os.environ["WEAVE_SYSTEM_CONFIG_USE_LWIP"] == "1":
+            print hred("WARNING: Test skipped due to LwIP-based network cofiguration!")
+            return
+
+        options = happy.HappyNodeList.option()
+        options["quiet"] = True
+
+        self.__run_inet_dns_test("node01")
+
+
+    def __run_inet_dns_test(self, node):
+        options = WeaveInetDNS.option()
+        options["quiet"] = False
+        options["node_id"] = node
+        options["tap_if"] = "wlan0"
+        options["node_ip"] = "10.0.1.3"
```

Comment:
hmm??

Reply:
We could specify the argument here as a prefix and then call getNodeAddressesOnPrefix or just call getNodePublicIPv4Address if we only expect one IPv4 address.

Reply:
1. getNodeAddressesOnPrefix will need prefix as argument
2. seems getNodePublicIPv4Address will not work for this case, because public ip need network type as "WAN"
seems getNodeInterfaceAddresses() will work better? I will test it out

Reply:
That `getNodeAddressesOnPrefix` needs an argument should not pose a problem. We can pass the prefix as an argument. The topology is known and fixed for this test and the prefix is fixed for the topology.

Reply:
sure, will pass in prefix as argument then